9armbot/9armbot.js

```diff
@@ -336,42 +318,38 @@ client.on('message', (channel, tags, message, self) => {
     if (marketOpen || isSubscriber(tags)) {
         /* query amount of coin */
         if (message == '!coin') {
-            let _player = player.getPlayerByUsername(tags.username)
-            if (_player)
-                client.say(channel, `@${_player.username} มี ${_player.coins} armcoin.`);
-            else
-                client.say(channel, `@${tags.username} มี 0 armcoin.`);
+            let _player = player.getOrCreatePlayer(tags.username);
+            client.say(channel, `@${_player.username} มี ${_player.coins} armcoin.`);
             return;
         }
 
 
         /* usage: !gacha [amount] */
         /* We are trying to control the inflation. The return, on average should be a loss for users. */
-        let gacha_re = /^!gacha\s*(\d*)/;
+        let gacha_re = /^!gacha\s([0-9]+)/;
         let group = message.match(gacha_re);
         if (group) {
-            if (!group[1])
-                gacha(channel, tags, 1);
-            else
-                gacha(channel, tags, parseInt(group[1]));
+            let amt = Number(group[1]);
```

Comment:
this will not let user do `!gacha` to do gacha for one coin. I agree that the original logic is kinda suck but I will not approve this.

Reply:
I have a plan for common function that will deal with regex and extract numbers for bot commands. Will update if I have time.

Reply:
Fixed